src/box/lua/console.lua

```diff
@@ -662,6 +687,15 @@ end
 -- Print result to stdout
 --
 local function local_print(self, output)
+    if output:endswith(output_eos["lua"]) then
+        local new_eos
+        if (self.remote ~= nil) then
+            new_eos = self.remote.local_lua_eos
+        else
+            new_eos = self.local_lua_eos
+        end
+        output = output:gsub(output_eos["lua"], new_eos)
```

Comment:
`gsub()` is not a good idea here due to a couple of reasons:

* It'll replace all occurences of given string, not the last one.
* If the symbol is special for Lua regexp syntax (for example, `.`), the result will be even more strange (let's try `console.eos('.')`).

Reply:
It's OK to gsub with `'.'`:
```
tarantool> ('abc'):gsub('b', '.')
---
- a.c
- 1
...
```
The `output_eos["lua"]` is non-configurable server-side EOS, we are guaranteed to have only one occurrence of it, because `socket:read` reads until it.

Reply:
> The `output_eos["lua"]` is non-configurable server-side EOS

https://www.tarantool.io/en/doc/latest/reference/reference_lua/console/#lua-function.console.eos

Reply:
> > The `output_eos["lua"]` is non-configurable server-side EOS
> 
> https://www.tarantool.io/en/doc/latest/reference/reference_lua/console/#lua-function.console.eos

You are right, the `output_eos` can be modified, but still, as it is an End Of Stream, we a guaranteed that it occurs only once.

Reply:
> ```lua
> ('abc'):gsub('b', '.')
> ```

We're discussing changing of a server side EOS. So your example should be like so:

```lua
('abc.'):gsub('.', '')
```

Just cut last N symbols and append the new EOS. We don't need a regex engine for this task.

```lua
output = output:sub(1, -#output_eos["lua"] - 1) .. new_local_eos
```

Reply:
You are right. Fixed.